src/OrchardCore.Modules/OrchardCore.Email.Azure/Views/AzureEmailSettings.Edit.cshtml

```diff
@@ -0,0 +1,52 @@
+@using Microsoft.AspNetCore.Mvc.Localization
+@using OrchardCore.Email.Azure.ViewModels
+
+@model AzureEmailSettingsViewModel
+
+<div class="mb-3">
+    <div class="form-check">
+        <input type="checkbox" class="form-check-input" asp-for="IsEnabled" checked="@Model.IsEnabled" role="button" data-bs-toggle="collapse" data-bs-target="#Azure_Provider_Settings" aria-expanded="@Model.IsEnabled.ToString().ToLowerInvariant()" aria-controls="Azure_Provider_Settings" />
+        <label class="form-check-label" asp-for="IsEnabled">@T["Enable Azure Provider"]</label>
```

Comment:
Because if you enable the feature but it is not configured, then it will/should not work. If a configuration is provided via appsettings, then the box is checked by default.

Reply:
This is what I commented initially here: https://github.com/OrchardCMS/OrchardCore/pull/15254#issuecomment-1934708267

IMO what should change overall:

- Drop the checkboxes for enabling the providers, it should just be feature-based.
- Move the SMTP provider to a separate feature.
- When the email feature is enabled, but there are no providers with valid settings, display an admin warning.
- The last registered provider with valid settings should be picked up automatically as the default. (Zoltán commented this in a separate thread I think)

Reply:
It's OK to require the feature to be configured, but I don't see the reason to have a separate checkbox too. To put it otherwise, having an internal flag of `IsConfigured` or `IsUsable` is fine, of course, but I don't think it should be surfaced on the UI as user-editable.

As for the separate feature, I'm on the fence. I don't like bolting on features that we wouldn't have coded in the same way originally. Like, if we'd have started with SMTP and ACS, I guess we'd created and `Email`, `Smtp`, and `Email.Azure` module, as Hisham did under the other PR. Now the Email module is kind of in between, providing the base infra and a default provider too (like Media, but unlike Search). That would make the change breaking though.

And yeah, I'd also display an admin warning persistently, not just on the Email settings page.

Reply:
> It's OK to require the feature to be configured, but I don't see the reason to have a separate checkbox too. To put it otherwise, having an internal flag of IsConfigured or IsUsable is fine, of course, but I don't think it should be surfaced on the UI as user-editable.

If there is no checkbox, the user will have no way to disable it if they want to. The idea is to give the use full control "regardless of the configuration in the settings provider". Like you could provide Azure settings but the default, but the user say I don't want to use it and I want to use custom SMTP settings.

Reply:
Well, they can just not enable it :). Currently, you have to enable the Azure module, then also enable it with the checkbox. I don't see a use case for when you want to enable the module but don't want to enable the provider. It's a second switch for the same thing.

Note how we're also talking about putting the SMTP provider into its own feature (and module).

Reply:
I removed the checkbox from the Azure communication since that belongs to a separate feature. 

> Note how we're also talking about putting the SMTP provider into its own feature (and module).
This is a breaking change and I don't think we should do it. With the current setup, you have full control and it is fully backward compatible.

Reply:
Taking a step back, if you were to start designing OC's Email feature, knowing that we'll need these two implementations, how would you structure it? What would be the modules?

Reply:
I would put the SMTP provider into separate module "not just feature" just because it has dependency on external project "MailKit". I would also include 3 different providers 
 1. System provider: "uses settings that are defined in the configuration provider"
 2. SMTP provider: standard SMTP providers
 3. File provider: this one will be the same as the SMTP but would deliver messages into the folder. 

The above approach will follow the same design we have in the search module.

Reply:
I see, thanks. I think about the structuring in a similar way, though I don't think I'm following regarding the 3 providers. I'd bite the partially breaking bullet here. So how about this?

- Move what's in the Email module and only related to SMTP to a new `Email.Smtp` module. The configuration provider keys would change accordingly.
- Add a migration to the Email module that enables the Smtp feature if either:
  - `SmtpSettings` in site settings exists and doesn't have a new property set (a property added just for detecting this 1.8-1.9 upgrade), or
  - the old configuration keys are used.
- In the Smtp module, support the old configuration keys too.

I think this is good enough. If as a site owner you want to be diligent, then you'll need to change your recipes (if you enabled `Email` before, now enable `Email.Smtp`) and your configuration keys, but you don't have to and things will keep working. The only scenario I see this actually breaking is if your app didn't use at least the `Core` meta package to load all modules but you added module package individually, and now you need to add a new package.

Or, just document the breaking change like Hisham did or like Jasmin for Elasticsearch, of course.

Reply:
Okay. If we are going through and creating a new module for SMTP, I would like to change how the configuration provider works. The SMTP provider would only work with the admin settings. Then, we'll introduce preconfigured provider named "System" or any other name. The System provider will use SMTP service with a settings that are provided by configuration provider. This way we have two separate providers that use SMTP. The system provider will not have a tab since it is only available if the settings exists. But the user will be able to change the default provider to "SMTP", "System" or "Azure" assuming all three are configured. If one provider settings the "System" will be available otherwise it will never show up. If the user enabled SMTP via admin settings then they would also have that. Have a system provider like that is very useful because some SaaS owners could enable that by default for their tenants and then each tenant can switch that to SMTP or Azure if they want to.

Reply:
Sounds good, my only issue is about it being hard to grasp what that "System" does (and that it's a different name for the same underlying SMTP technology configured differently).

If I as a technical user who wants to manage e-mail sending from the admin (like creating Workflows) but am not involved with hosting (so I don't know/care about SendGrid, ACS, or anything similar) then I should be only concerned with "this option will let me send e-mail in a way that the IT guys set up". However, this can be implemented with ACS too.

So, if you'd like to achieve this, I'd rather make it configurable from the env which provider is the System one.

Reply:
Yes system may not be a good name. We can call it "Default". Default is an email provider that is configured by default using configuration providers via `OrchardCore_Email` section. This will allow us to expose it only when it is configured. This way the user would use Default or any other provider. Note, that we'll internally use SMTP service to handle the default configuration but the user will/should not care about the implementation details or the configuration.

So in another words, if you configure your email currently using `OrchardCore_Email` section, then you'll see "Default" provider. And we'll register it last so that way if someone is already using this configuration, we'll just load it since it is the last registered provider. "unless the user specify an options".

I'll try to address these feedback soon.